### PR TITLE
Introduce `TxOutExt`

### PR DESCRIPTION
**Pull request by tcharding**
This one is pretty easy. Leaves the actual move of `TxOut` for later.

### CODE REVIEW

File: bitcoin/src/blockdata/transaction.rs [hunk 1]
```diff
@@ -387,7 +387,7 @@ mod tmp {
         /// To use a custom value, use [`minimal_non_dust_custom`].
         ///
         /// [`minimal_non_dust_custom`]: TxOut::minimal_non_dust_custom
-        pub fn minimal_non_dust(script_pubkey: ScriptBuf) -> Self {
+        pub fn minimal_non_dust(script_pubkey: ScriptBuf) -> TxOut {
```
Comment (Kixunil) [inline bitcoin/src/blockdata/transaction.rs]: Why is this needed? The extension trait is implemented for `TxOut` so `Self` refers to `TxOut` thus it shouldn't be needed.

Comment (tcharding) [reply bitcoin/src/blockdata/transaction.rs]: Oh I have no idea why I had it in mind that doing this was part of creating the extension traits. Removed.

**Review by Kixunil — APPROVED** [review]
ACK b61adf7ca4fefd6c5fe3879de7baecf1ffde9d9b

**Comment by yancyribbens** [comment]
What's the purpose of this?

**Comment by Kixunil** [comment]
Same as all other extension trait changes. To move POD types to `primitives` easily.

**Review by apoelstra — APPROVED** [review]
ACK b61adf7ca4fefd6c5fe3879de7baecf1ffde9d9b successfully ran local tests
